File: typescript/src/party.ts
import { Party, SignFunction } from '../types'
import createKeyExchange from './key-exchange'
import createSafetyNumber from './safety-number'

const createParty = (
  isInitiator: boolean,
  sign: SignFunction,
  identityPublicKey: Uint8Array
): Party => ({
  calculateSafetyNumber: createSafetyNumber(identityPublicKey),
  performKeyExchange: createKeyExchange(isInitiator, sign, identityPublicKey)
})

export default createParty

File: rust/src/party.rs
use crate::key_exchange::create_key_exchange;
use crate::safety_number::create_safety_number;
use crate::types::{Bytes, Party, SignFunction};

pub fn create_party<'a>(
    is_initiator: bool,
    sign: &'a SignFunction,
    identity_public_key: &'a Bytes,
) -> Party<'a> {
    Party {
        calculate_safety_number: create_safety_number(identity_public_key),
        perform_key_exchange: create_key_exchange(is_initiator, sign, identity_public_key),
    }
}
